src/jit/optcse.cpp

```diff
@@ -2072,6 +2073,25 @@ class CSE_Heuristic
         Compiler::CSEdsc*      dsc = successfulCandidate->CseDsc();
         Compiler::treeStmtLst* lst;
 
+        // If there's just a single def for the CSE, we'll put this
+        // CSE into SSA form on the fly. We won't need any PHIs.
+        bool     cseInSSA  = false;
+        unsigned cseSsaNum = SsaConfig::RESERVED_SSA_NUM;
+
+        if (dsc->csdDefCount == 1)
+        {
+            JITDUMP("CSE #%02u is single-def, so associated cse temp V%02u will be in SSA\n", dsc->csdIndex,
+                    cseLclVarNum);
+            cseInSSA                                    = true;
+            m_pCompiler->lvaTable[cseLclVarNum].lvInSsa = true;
+
+            // Allocate the ssa num. We'll fill in block and tree when we find the def below.
+            BasicBlock*   defBlock  = nullptr;
+            GenTree*      defTree   = nullptr;
+            CompAllocator allocator = m_pCompiler->getAllocator(CMK_SSA);
+            cseSsaNum = m_pCompiler->lvaTable[cseLclVarNum].lvPerSsaData.AllocSsaNum(allocator, defBlock, defTree);
```

Comment:
There should be no need to specify the block and tree when allocating the SSA number. The 2 parameters exists solely for convenience in cases where you happen to have them already.